File: frontend/src/features/actividades/validar_condiciones_desembolso/models/validar_condiciones_desembolso.ts
import type { Auditoria } from '@/models/Auditoria';

export interface ValidarCondicionesDesembolso extends Auditoria {
  id: number;
  id_expediente: number;
  id_actividad: string;
  origen_tramite: string | null;
  confirmar_plan_pagos: boolean;
  requiere_escalamiento_comercial: string | null;
  suspendida: boolean;
  conteo_caidas: number;
  observaciones: string | null;
}

export interface GetByExpedienteResponse {
  formulario: ValidarCondicionesDesembolso;
  puede_suspender: boolean;
  origen_tramite: string;
  conteo_caidas: number;
}

export const EMPTY_VALIDAR_CONDICIONES_DESEMBOLSO = (id_expediente: number): ValidarCondicionesDesembolso => ({
  id: 0, id_expediente, id_actividad: 'BBVA_ESCRITURACION_VALIDAR_CONDICIONES_DESEMBOLSO',
  origen_tramite: null, confirmar_plan_pagos: false, requiere_escalamiento_comercial: null,
  suspendida: false, conteo_caidas: 0, observaciones: null,
  is_active: true, row_status: true, created_by: 0, created_date: '', modified_by: null, modified_date: null,
});

File: backend/Multibanca.Domain.Models/Multibanca/BBVA/Escrituracion/validar_condiciones_desembolso.cs
namespace Multibanca.Domain.Models.Multibanca.BBVA.Escrituracion;

public class validar_condiciones_desembolso
{
    public long     id { get; set; }
    public long     id_expediente { get; set; }
    public string?  id_actividad { get; set; }
    public string?  origen_tramite { get; set; }
    public bool     confirmar_plan_pagos { get; set; }
    public string?  requiere_escalamiento_comercial { get; set; }
    public bool     suspendida { get; set; }
    public int      conteo_caidas { get; set; }
    public string?  observaciones { get; set; }

    public bool      is_active { get; set; }
    public bool      row_status { get; set; }
    public int       created_by { get; set; }
    public DateTime  created_date { get; set; }
    public int?      modified_by { get; set; }
    public DateTime? modified_date { get; set; }
}
